flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/dataset/DataSetMap.scala

```diff
@@ -63,7 +63,7 @@ class DataSetMap(
       config: TableConfig,
       expectedType: Option[TypeInformation[Any]])
     : DataSet[Any] = {
-    val inputDataSet = input.asInstanceOf[DataSetRel].translateToPlan(config)
+    val inputDataSet = input.asInstanceOf[DataSetRel].translateToPlan(config, expectedType)
```

Comment:
You cannot forward the expected type, since you don't know what the previous operator does.
E.g. if the expected type is `Tuple2` but the previous operator outputs records with one field.
Why did you change this call?